pages/partials/questionFooter.ejs

```diff
@@ -90,34 +90,22 @@
       </div>
       <% } %>
     </form>
-
-    <script>
-      $('form').on('submit', function(e) {
-          if (!$(this).data('submitted')) {
-              $(this).data('submitted', true);
-
-              var form = $(this);
-              form.find('.disable-on-submit').each(function (index) {
-                  // Create a disabled clone of the submit button
-                  $(this).clone(false).removeAttr('id').prop('disabled', true).insertBefore($(this));
-
-                  // Hide the actual submit button and move it to the beginning of the form
-                  $(this).hide();
-                  form.prepend($(this));
-              });
-
-          } else {
-              e.preventDefault();
-          }
-      });
-
-      $('a.disable-on-click').on('click', function() {
-          $(this).addClass('disabled');
-      });
-    </script>
   </div>
 <% } %>
-
+<script>
+    // Instead of disabling buttons upon form submit (2e0b8eb and #2187),
+    // disable the form upon button clicks to resolve issue with
+    // externally graded questions (#2234).
+    $(document).ready(() => {
+        const form = $('form.question-form');
+        $('button.disable-on-submit').click(function() {
```

Comment:
I worry a bit that if something interferes with JS execution (after we've installed this onclick handler) then we'll be unable to submit the form. But I think that even if a question contains bad JS, this handler will continue to run? Is it possible to test robustness to JS errors inside questions?

Reply:
In general it's probably fine (but probably still browser-dependent): https://stackoverflow.com/a/29394912/13138364

> An uncaught runtime error only stops the current execution, which may be
>
> * the execution of a script
> * the call of an event handler
>
> Suppose you have a runtime error while handling an event, the only problem you might have (apart not really handling the event) is a non consistent state of your user variables if your event handler modifies some of them. Other event handlers won't be impacted besides that.
>
> So it can usually be considered as non catastrophic (I guess I don't have to remember it's a good practice to fix errors anyways and that flooding the console with errors isn't a good thing).

Reply:
I tested adding a broken script above the double-submit script:
```javascript
<script>
    alert('malformed alert
</script>
```

* [x] Firefox: still works
* [x] Chrome: still works
* [ ] Edge: does **NOT** work

Actually upon further testing, I'm not sure this solution works in Edge even without the broken script. It was working yesterday, but now it's giving me `unknown __action: undefined` 😩

Reply:
Huh! That's weird that Edge is different. The `unknown __action: undefined` means the server got a POST with the `body` containing `__action: undefined`. Normally `__action` should be set by the submit button `value`: https://github.com/PrairieLearn/PrairieLearn/blob/master/pages/partials/questionFooter.ejs#L10

Reply:
Interesting, I don't know what was mentioned during OH today since I was working on other things, but my thoughts here are that `__action` only gets set when the form gets submitted through the button, and when the form gets submit via js there's no `__action` value set.  Maybe you can append a hidden input tag to get around this?

Reply:
With some hackery, the hidden `input` fixes the POST issue on Edge... but unfortunately the form disabling behavior is somehow totally ignored on Edge. I'm not even sure how/why 😩

Reply:
Do you have to set the button to have `type=“button”`?  I think by default it will “submit” the page.

Reply:
You mean to prevent the bound `button` from "submitting" along with the JS `form.submit()`?

Reply:
Right, unless I'm mistaken the default behaviour is to submit the form on button press?

Reply:
I'd been hoping to have the debounce code function as progressive enhancement, so the page would still work without JS running. Of course this might not be possible, given the weird behavior from Edge.

Reply:
Are you talking about the old Edge or the new Chrome-based Edge?

If Edge doesn't respect disabling form elements directly, would it work to layer a semitransparent div on top of the controls to "disable" them, similar to the annoying nag screens that news media pop up?

Reply:
Seems that I still have the non-Chromium Edge for some reason. Windows Update is up-to-date, and I'm reading that Chromium-Edge's public release was Jan 2020, so not sure why I'm still on v44.

Reply:
Based on the original v2 solution where cloned buttons were not properly removed on externally graded questions, the same would happen with an overlay. It looks like we'll have to go full circle and debug that original problem.

Reply:
I'm still hoping that the original problem is not too hard to debug (famous last words). Let me know if you need an extra set of eyes on it.